tools/bpf2c/bpf2c.cpp

```diff
@@ -291,9 +291,14 @@ main(int argc, char** argv)
 
         // Parse per-program data.
         for (const ebpf_api_program_info_t* program = infos; program; program = program->next) {
+            // Skip if a subprogram.  A subprogram is defined by libbpf as any
+            // program in the .text section when multiple programs exist.
+            if (!strcmp(program->section_name, ".text") && infos->next != nullptr) {
```

Comment:
question: does subprogram here mean the bpf-to-bpf functions? (which are not the entry BPF programs)? are they always guaranteed to be in ".text" section?

Reply:
Libbpf only treats them as such if they are in the ".text" section.  So apparently.